volatility3/framework/plugins/windows/psscan.py

```diff
@@ -261,40 +261,62 @@ def _generator(self):
             filter_func=pslist.PsList.create_pid_filter(self.config.get("pid", None)),
         ):
             file_output = "Disabled"
-            if self.config["dump"]:
-                # windows 10 objects (maybe others in the future) are already in virtual memory
-                if proc.vol.layer_name == kernel.layer_name:
-                    vproc = proc
+
+            # windows 10 objects (maybe others in the future) are already in virtual memory
+            # if the proc is built on the same layer as the kernel then it is already
+            # in 'virtual' memory.
+            if proc.vol.layer_name == kernel.layer_name:
+                # proc is already in a virtual mem, so a new object is not needed. it means
+                # that if physical addresses are requested in the output then proc.vol.offset
+                # cannot be used because it will be virtual, so the mapping is needed.
+                vproc = proc
+                if self.config["physical"]:
+                    # the display should be physical addresses, so proc cannot be used. The
+                    # mappings are needed to find where it would be physically.
+                    offset = (_, _, offset, _, _) = list(
+                        memory.mapping(offset=proc.vol.offset, length=0)
+                    )[0]
                 else:
-                    try:
-                        vproc = self.virtual_process_from_physical(
-                            self.context,
-                            kernel.layer_name,
-                            kernel.symbol_table_name,
-                            proc,
-                        )
-                    except exceptions.PagedInvalidAddressException:
-                        vproc = None
+                    # the display should be virtual addresses, so proc can be used
+                    offset = proc.vol.offset
 
-                file_output = "Error outputting file"
-                if vproc:
-                    file_handle = pslist.PsList.process_dump(
-                        self.context,
-                        kernel.symbol_table_name,
-                        pe_table_name,
-                        vproc,
-                        self.open,
-                    )
+                #  renderers.UnreadableValue()
+            else:
+                # proc is in virtual mem, so a new object needs to be creatd.
+                vproc = self.virtual_process_from_physical(
```

Comment:
This is no longer wrapped in a handler for `InvalidAddressExceptions`, will that cause a problem if it goes wrong?